plugins/async-node/core/src/index.test.ts

```diff
@@ -37,6 +38,90 @@ const basicFRFWithActions = {
   },
 };
 
+const asyncNodeTest = async (resolvedValue: any, expectedActionType: string) => {
+  const plugin = new AsyncNodePlugin({
+    plugins: [new AsyncNodePluginPlugin()]
+  });
+
+  let deferredResolve: ((value: any) => void) | undefined;
+
+  plugin.hooks.onAsyncNode.tap('test', async (node: Node.Node) => {
+    return new Promise((resolve) => {
+      deferredResolve = resolve;
+    });
+  });
+
+  let updateNumber = 0;
+
+  const player = new Player({ plugins: [plugin] });
+
+  player.hooks.viewController.tap('async-node-test', (vc) => {
+    vc.hooks.view.tap('async-node-test', (view) => {
+      view.hooks.onUpdate.tap('async-node-test', () => {
+        updateNumber++;
+      });
+    });
+  });
+
+  player.hooks.viewController.tap('async-node-test', (vc) => {
+    vc.hooks.view.tap('async-node-test', (view) => {
+      view.hooks.resolver.tap('async-node-test', () => {
+        updateNumber++;
```

Comment:
what is the `updateNumber`? Originally I had intended for `updateNumber` to represent how many times the tree is resolved and an update is provided for the UI, but if we're updating `updateNumber` here, then it's not longer following the same semantic intent.

Reply:
following our slack conversation, this resolver is supposed to be used to force a third update through the `update` function, forcing the tree to be resolved again to make sure the replacement of an async node with another one is "safe"

In that process, I believe `onAsyncNode` should be called again, but it won't go to the users

Reply:
done